src/vivarium/framework/engine.py

```diff
@@ -399,9 +407,5 @@ def run_simulation(
     simulation = SimulationContext(
         model_specification, components, configuration, plugin_configuration
     )
-    simulation.setup()
-    simulation.initialize_simulants()
-    simulation.run()
-    simulation.finalize()
-    simulation.report()
+    simulation.run_simulation()
```

Comment:
It kind of annoys moe that this method is the same name as the function that calls it but I can't think of anything better (I tried using a simulation.run() and then renaming the old simulation.run() to simulation.take_steps(), but that's too user-facing b/c of IteractiveContext)

Reply:
Would it be cleaner to get rid of this `run_simulation` method and just create the SimulationContext in the cli method and call sim.run_simulation there?

Reply:
I thought about it but kinda liked the idea that this entry point instantiated the object and then ran it. But looking at it now I don't like it so much

Reply:
Though whoever wrote the docs went out of the way to discuss how "the combination of initializing and running the simulation is encpsulated in the `run_simulation` command..." ([reference](https://github.com/ihmeuw/vivarium/blob/main/docs/source/concepts/entry_points.rst))

We should chat about whether this makes sense or not